src/main/kotlin/com/sourcegraph/cody/chat/ui/LLMModelDropdown.kt

```diff
@@ -0,0 +1,73 @@
+package com.sourcegraph.cody.chat.ui
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.invokeLater
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBox
+import com.intellij.ui.MutableCollectionComboBoxModel
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.ChatModelsParams
+import com.sourcegraph.cody.agent.protocol.ChatModelsResponse
+import com.sourcegraph.cody.chat.SessionId
+import com.sourcegraph.cody.config.CodyAccount.Companion.isEnterpriseAccount
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.ui.ChatModel
+import com.sourcegraph.cody.ui.LLMModelComboBoxItem
+import com.sourcegraph.cody.ui.LLMModelComboBoxRenderer
+import com.sourcegraph.common.BrowserOpener
+import java.util.concurrent.TimeUnit
+
+class LLMModelDropdown(val project: Project, initiallySelected: ChatModel?) :
+    ComboBox<LLMModelComboBoxItem>(MutableCollectionComboBoxModel()) {
+
+  init {
+    val activeAccountType = CodyAuthenticationManager.instance.getActiveAccount(project)
+    if (activeAccountType.isEnterpriseAccount()) {
```

Comment:
I'm wondering, since this is not blocking, if it's guaranteed to be already properly initialised when `LLMModelDropdown` instance is created.

Reply:
We need `ChatPanel` to have `LLMDropdown`
and we need `CodyAgentSession` to have `ChatPanel`
and we need an account to have the session. 

Neither `getActiveAccount` nor `isEnterpriseAccount` is blocking.